hydra-node/src/Hydra/Node.hs

```diff
@@ -167,7 +173,7 @@ processEffect HydraNode{hn, oc, server, eq, env = Environment{party}} tracer e =
     ClientEffect i -> sendOutput server i
     NetworkEffect msg -> broadcast hn msg >> putEvent eq (NetworkEvent msg)
     OnChainEffect tx -> postTx oc tx
-    Delay after event -> void . async $ threadDelay after >> putEvent eq event
+    Delay after event -> putEventAfter eq after event
```

Comment:
This is nice. I wonder why we even have the `Delay` now. What about

```haskell
processNextEvent HydraNode{hh, env} e =
  modifyHeadState hh $ \s ->
    case Logic.update env (ledger hh) s e of
      NewState s' effects -> (NewState s' effects, s')
      Error err -> (Error err, s)
      Wait -> (Wait, s)
```

```haskell
stepHydraNode tracer node@HydraNode{eq, env = Environment{party}} = do
  e <- nextEvent eq
  traceWith tracer $ ProcessingEvent party e
  atomically (processNextEvent node e) >>= \case
    NewState _ effs -> forM_ effs (processEffect node tracer) >> traceWith tracer (ProcessedEvent party e)
    Error err -> traceWith tracer (ErrorHandlingEvent party e err)
    Wait -> putEventAfter eq 0.1 e
```

And remove this line as well as the `Delay` effect altoghether? :slightly_smiling_face:  :tada:

Reply:
I actually got confused for a bit initially. Seeing that we have `Wait` as an outcome, but a `Delay` effect. Yet, we only use the `Delay` effect to re-enqueue events which is exactly the semantic of `Wait`! Hence, I guess you're right :)

Reply:
The point was to be able to renqueue a _different_ event IIRC

Reply:
We had only a `Wait` before and did what @ch1bo suggested

Reply:
Ah yeah.. there is the ShouldPostFanout as well. Fine, maybe we'll get rid of some of these things later.. or not.